SpatialGDK/Extras/insights-proxy/InsightsProxy.py

```diff
@@ -0,0 +1,37 @@
+import socket 
+import sys 
+
+def main():
+    is_connected = False
+    while not is_connected:
+        try:
+            print("Attempting connection to 127.0.0.1:1981 [server worker]")
+            worker_connection = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            worker_connection.connect(("127.0.0.1", 1981))
+            worker_connection.settimeout(1.0)
+            is_connected = True
+        except socket.error:
```

Comment:
Do we need some timeout with an exception in here so this doesn't spin forever?

Reply:
I think spinning should be ok, this is a manual process and I sort of leave it up to the developer to ensure the port is accessible and listening (triggered from Insights) - what are your thoughts?